[aot_inductor] added a utility function aoti_torch_print_tensor_handle

chenyang78 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #120660

Added a function to print tenosr values for a tensor handle.
It can be injected to the cpp wrapper code and help debug
numerical issues.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames

Commit:
[aot_inductor] added a utility function aoti_torch_print_tensor_handle

Added a function to print tenosr values for a tensor handle.
It can be injected to the cpp wrapper code and help debug
numerical issues.

[ghstack-poisoned]

Commit:
Update on "[aot_inductor] added a utility function aoti_torch_print_tensor_handle"


Added a function to print tenosr values for a tensor handle.
It can be injected to the cpp wrapper code and help debug
numerical issues.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 kadeng muchulee8 aakhundov ColinPeppler amjames

[ghstack-poisoned]

Commit:
Update on "[aot_inductor] added a utility function aoti_torch_print_tensor_handle"


Added a function to print tenosr values for a tensor handle.
It can be injected to the cpp wrapper code and help debug
numerical issues.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 kadeng muchulee8 aakhundov ColinPeppler amjames

[ghstack-poisoned]

Commit:
Update on "[aot_inductor] added a utility function aoti_torch_print_tensor_handle"


Added a function to print tenosr values for a tensor handle.
It can be injected to the cpp wrapper code and help debug
numerical issues.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 kadeng muchulee8 aakhundov ColinPeppler amjames

[ghstack-poisoned]

desertfire commented on torch/csrc/inductor/aoti_torch/c/shim.h:
```diff
@@ -463,6 +463,10 @@ AOTI_TORCH_EXPORT AOTITorchError aoti_torch_view_dtype(
 
+AOTI_TORCH_EXPORT void aoti_torch_print_tensor_handle(
+    AtenTensorHandle self,
+    const char* msg = nullptr);
```
C interface doesn't allow default value?

chenyang78 replied:
> C interface doesn't allow default value?

Good point. Fixed. Thanks.

desertfire commented on torch/csrc/inductor/aoti_torch/shim_common.cpp:
```diff
@@ -789,6 +789,13 @@ AOTI_TORCH_EXPORT AOTITorchError aoti_torch_view_dtype(
+    AtenTensorHandle self,
+    const char* msg) {
+  at::Tensor* t = reinterpret_cast<at::Tensor*>(self);
+  std::cout << "[" << msg << "]:" << *t << "\n";
```
If `msg` is nullptr, the text after `msg` will be dropped.

chenyang78 replied:
Fixed. Thanks.

Commit:
Update on "[aot_inductor] added a utility function aoti_torch_print_tensor_handle"


Added a function to print tenosr values for a tensor handle.
It can be injected to the cpp wrapper code and help debug
numerical issues.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 kadeng muchulee8 aakhundov ColinPeppler amjames

[ghstack-poisoned]

desertfire commented on torch/csrc/inductor/aoti_torch/shim_common.cpp:
```diff
@@ -789,6 +789,17 @@ AOTI_TORCH_EXPORT AOTITorchError aoti_torch_view_dtype(
+AOTI_TORCH_EXPORT void aoti_torch_print_tensor_handle(
+    AtenTensorHandle self,
+    const char* msg) {
+  at::Tensor* t = reinterpret_cast<at::Tensor*>(self);
```
Let's use `tensor_handle_to_tensor_pointer` here like in other places.

Commit:
Update on "[aot_inductor] added a utility function aoti_torch_print_tensor_handle"


Added a function to print tenosr values for a tensor handle.
It can be injected to the cpp wrapper code and help debug
numerical issues.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 kadeng muchulee8 aakhundov ColinPeppler amjames

[ghstack-poisoned]

chenyang78 commented:
@pytorchbot merge

chenyang78 commented:
@pytorchbot merge
